.github/workflows/codeql-analizer.yml

```diff
@@ -0,0 +1,16 @@
+name: CodeQL Analysis
+on: [pull_request]

```

Comment:
I have had some issues to find and run this task via [`act`](https://github.com/nektos/act). 
It works with the `on` section as in https://github.com/cosmos/cosmos-sdk/blob/master/.github/workflows/codeql-analysis.yml
Also `name` and `permissions` seem reasonable settings.